### PR TITLE
Add getElementSpeed function

### PR DESCRIPTION
**Pull request by Pirulax**
Just a simple getElementSpeed function which was already available as a useful function, but it's much faster if it's written in cpp, also, it can be used everywhere, unlike the useful function(I mean, having this in the source code reduces redundancy).
Also, the same arguments can be passed as to setElementSpeed, so no scripts will be broken(actually, functions defined inside scripts will override the default ones in MTA).
Wanted to do setElementSpeed too, but when I tested the useful function from the Wiki it produced some weird things, so added that as todo.

### CODE REVIEW

**Comment by CrosRoad95** [comment]
Should return negative value if vehicle reversing
https://wiki.multitheftauto.com/wiki/IsVehicleReversing

**Comment by Pirulax** [comment]
Does you speedo goes in the opposite direction when you go backwards?

**Comment by CrosRoad95** [comment]
nope

**Comment by qaisjp** [comment]
Speed should always be positive but I'm not convinced that this function is necessary. Maybe we should rename "useful functions" to "useful snippets"...

**Comment by Pirulax** [comment]
Actually, haven't tested the performance of the Lua one, but im sure this one has much better performance.

**Comment by patrikjuvonen** [comment]
I think it would help your case if you provided us with a proper benchmark and its results. If you can't prove your case it doesn't make sense to say it does x thing when it doesn't and the other way around.

**Comment by Pirulax** [comment]
Okay,I'll post some benchmarks later.

**Comment by samr46** [comment]
The difference in performance is probably small. Basically because your function uses the same algorithm and existing [getElementVelocity](https://wiki.multitheftauto.com/wiki/GetElementVelocity).
I'm pretty sure that [getElementVelocity](https://wiki.multitheftauto.com/wiki/GetElementVelocity) can't get velocity of moving object ([moveObject](https://wiki.multitheftauto.com/wiki/MoveObject)). I think you could modify this part. It may be useful.

File: Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 1]
```diff
@@ -491,6 +491,46 @@ bool CStaticFunctionDefinitions::GetElementVelocity(CClientEntity& Entity, CVect
     return true;
 }
 
+int CStaticFunctionDefinitions::GetElementSpeed(CClientEntity& Entity, unsigned char ucUnit, float& fSpeed)
```
Comment (xerox8521) [inline Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp:494]: Return type should be bool

**Comment by qaisjp** [comment]
I don't think this function is necessary and @samr46's comment about performance is probably correct.

If another developer feels otherwise then please do reopen.

**Pull request closed without merging**